pkg/service/aws/oidc.go

```diff
@@ -0,0 +1,159 @@
+	return fmt.Sprintf(trustPolicTemplate, federatedPrefix, oidcUrl, oidcUrl)
+}
+
+//generateTrustPolicyDocument read the current policy document as a map, create new policy using the template stringa and convert that to map
+//finally append the new policy to Statement section of the current policy
+func generateTrustPolicyDocument(currentPolicyDoc, oidcproviderArn, oidcIssuer string) (string, error) {
+	decodedValue, err := url.QueryUnescape(currentPolicyDoc)
+	if err != nil {
+		return "", err
+	}
+
+	policyDoc := make(map[string]interface{})
+	err = json.Unmarshal([]byte(decodedValue), &policyDoc)
+	if err != nil {
+		return "", err
+	}
+
+	newPolicy := getTrustPolicyDocument(oidcproviderArn, oidcIssuer)
+
+	newDoc := make(map[string]interface{})
+	err = json.Unmarshal([]byte(newPolicy), &newDoc)
+	if err != nil {
+		return "", err
+	}
+	policyDoc["Statement"] = append(policyDoc["Statement"].([]interface{}), newDoc)
+
+	b, err := json.Marshal(policyDoc)
+	if err != nil {
+		return "", err
+	}
+	return string(b), nil
+}
+
+// fetch OIDC from the cluster and attach it to role policy
+
+func (a *awsController) ConnectClusterOIDCToTrustPolicy(ctx context.Context, req *proto.ConnectClusterOIDCToTrustPolicyRequest) (*proto.ConnectClusterOIDCToTrustPolicyResponse, error) {
+	//get the oidc endpoint from the cluster spec
+
+	roleName := config.Get().OpenIDRole
+	region := req.Region
+	accountName := req.AccountName
+	session, err := NewSession(ctx, region, accountName)
+
+	if err != nil {
+		return nil, err
+	}
+	eksClient := session.getEksClient()
+
+	cluster, err := getClusterSpec(ctx, eksClient, req.ClusterName)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get the cluster")
+	}
+
+	if *cluster.Status != eks.ClusterStatusActive {
+		a.logger.Info(ctx, "cluster is not active yet", "status", *cluster.Status)
+		return nil, fmt.Errorf("cluster is not active, curent state: %s", *cluster.Status)
+	}
+	if cluster.Identity == nil {
+		a.logger.Info(ctx, "cluster doesnt have identity", "identity", nil)
+		return nil, errors.New("cluster identity is nil")
+	}
+
+	issuer := *cluster.Identity.Oidc.Issuer
+
+	a.logger.Info(ctx, "cluster found", "issuer", issuer)
+
+	iamClient := session.getIAMClient()
+	accountId, err := session.getAccountId()
+	if err != nil {
+		a.logger.Error(ctx, "failed to get the account id", "error", err)
+		return nil, errors.Wrap(err, "getAccountid")
+	}
+
+	clusterUrl := strings.Split(issuer, "https://")[1]
+	providerArn := fmt.Sprintf("arn:aws:iam::%s:oidc-provider/%s", accountId, clusterUrl)
+	//check if we already created a open id oidcProvider
+	_, err = iamClient.GetOpenIDConnectProviderWithContext(ctx, &iam.GetOpenIDConnectProviderInput{
+		OpenIDConnectProviderArn: &providerArn,
+	})
+
+	if err != nil {
+		a.logger.Error(ctx, "failed to fetch open id provider", "error", err)
+
+		//TODO: where is this coming from ? @mani
+		thumbprint := "9e99a48a9960b14926bb7f3b02e22da2b0ab7280"
```

Comment:
Replicate functionality from https://github.com/gruntwork-io/kubergrunt/blob/master/eks/oidc.go to get thumbprint.